docker/app/src/Controller/GiftcardController.php

```diff
@@ -60,30 +62,57 @@ public function giftget($id): JsonResponse
     /** 
      * @Route("/giftcard/redeem/{id}", methods={"POST"})
      */
-    public function giftredeem($id): JsonResponse
+    public function giftRedeem($id, Request $request): JsonResponse
     {
-        #body money
-        #validacija
+        #u body ubaci money koliko treba oduzet
+        #validacija rjesi da validira dobro sve i da provjeri jeli neko unia -50 npr ili +50
         $firestore = new FirestoreClient([
-            'projectId' => self::projectId,
+            'projectId' => self::PROJECTID,
         ]);
 
+        $newNum = $request->toArray();
+
         $doc = $firestore->collection($_ENV['COLLECTION'])->document($id);
         $snapshot = $doc->snapshot();
 
+        $oldNum = $snapshot->data();
+        $oldNum = $oldNum['currency']['amount'];
+        
+        $newNum = $newNum['amount'];
+        
         if (!$snapshot->exists()) {
```

Comment:
Creating nest of if else is effective but pretty ugly to read and follow. You can do the same logic but much better.
For example, removing the else tag in this place. As you want that code to be executed if $snapshot exists, you have if condition for return the error message "This giftcard does not exist" and below the rest of the code.
Keep always in mind that code should be clean to read as possible. You will come back in some time on your code and you wont be able to read it :D